src/lattice.cpp

```diff
@@ -710,69 +710,87 @@ const
 std::array<int, 3>
 HexLattice::get_indices(Position r, Direction u) const
 {
-  // The implementation for HexLattice currently doesn't use direction
-  // information. As a result, we move the position slightly forward to
-  // determine what lattice index the particle is most likely to be in.
-  r += TINY_BIT * u;
+  // result variables
+  int ix{}, ia{}, iz{};
```

Comment:
[ES.10](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es10-declare-one-name-only-per-declaration): Declare one name per declaration. I would tend to go for the more traditional:
```C++
int ix = 0;
int ia = 0;
int iz = 0;
```

Reply:
Better yet, don't declare them until right when they're needed.  `ix` and `ia` can be initialized directly, i.e.
```c++
  int ix = std::floor(r_o.x / (0.5*std::sqrt(3.0) * pitch_[0]));
  int ia = std::floor(alpha / pitch_[0]);
```
and `iz` can be safely declared without initialization before it's used,
```c++
  int iz;
  if (is_3d_) {
    ...
  }
```

Reply:
Nice I like it. When the mesh isn't 3D we'll still want `iz` to be initialized to 0 though, right?

Reply:
Yeah, you're right.  `iz` should be initialized to 0